### PR TITLE
Use Traefik and Letsencrypt to support https

### PR DESCRIPTION
**Pull request by w-t-yang**
We adopt Traefik and Letsencrypt to provide supports for https in OPA

The new docker compose file is compatible with existing installation guide.

In order to enable https, users need to provide 2 extra environment variables
- SUPERBLOCKS_PROXY_ADMIN_EMAIL
- SUPERBLOCKS_PROXY_REPLICA_COUNT, default: 0, assign it to 1 to enable https proxy.

### CODE REVIEW

File: docker/compose.yaml [hunk 1]
```diff
@@ -16,16 +16,57 @@ x-environment: &environment
     __SUPERBLOCKS_AGENT_SERVER_URL: ${__SUPERBLOCKS_AGENT_SERVER_URL:-https://app.superblocks.com}
 
 services:
+  proxy:
+    # Configurable variables for traefik to set up https
+    # SUPERBLOCKS_PROXY_ADMIN_EMAIL
```
Comment (taha-au) [inline docker/compose.yaml:21]:
```suggestion
    # SUPERBLOCKS_LETSENCRYPT_EMAIL
```

File: docker/compose.yaml [hunk 2]
```diff
@@ -16,16 +16,57 @@ x-environment: &environment
...
+    # SUPERBLOCKS_PROXY_REPLICA_COUNT, default 0
+    # SUPERBLOCKS_LETSENCRYPT_DIR, default letsencrypt
+    image: traefik:v2.8
+    deploy:
+      mode: replicated
```
Comment (taha-au) [inline docker/compose.yaml:26]: Not certain if this is necessary. Let's remove if it isn't

File: docker/compose.yaml [hunk 3]
```diff
@@ -16,16 +16,57 @@ x-environment: &environment
...
+      - '--certificatesresolvers.letsencrypt.acme.storage=/${SUPERBLOCKS_LETSENCRYPT_DIR:-letsencrypt}/acme.json'
+      - '--certificatesresolvers.letsencrypt.acme.httpchallenge=true'
+      - '--certificatesresolvers.letsencrypt.acme.httpchallenge.entrypoint=web'
+    labels:
+      ### core config
+      - 'traefik.enable=true'
+
+      ### http to https redirect
+      - 'traefik.http.routers.http-catchall.entrypoints=web'
+      - 'traefik.http.routers.http-catchall.rule=hostregexp(`{host:.+}`)'
```
Comment (taha-au) [inline docker/compose.yaml:48]: this is okay in the case that there is a single externally reachable service running on the VM/machine. it's fine with me, just want to flag it. let's add this caveat as a comment.

**Review by taha-au — APPROVED** [review]
Approved. Eventually when we have a wrapper, we should probably parse out the port and domain so that we can do the traefik routing more intelligently, and allow the use of non-standard ports.

File: docker/compose.yaml (hunk 1 above)
Comment (gpoulios-sb) [reply docker/compose.yaml:21]: I have seen in some other software that supports letsencrypt that they do no ask for this and instead they give some default email. I don't think letsencrypt has ever used those emails for anything.

File: docker/compose.yaml (hunk 2 above)
Comment (w-t-yang) [reply docker/compose.yaml]: Removed

**Pull request merged**